### PR TITLE
[New Language Support] Added Java CDK

### PR DESCRIPTION
**Pull request by patmeiler**
Added support for Java CDK

### CODE REVIEW

File: cdk/java/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Welcome to your CDK Java project!
```
Comment (alexcasalboni) [inline cdk/java/README.md:1]: I would stick to the same title patterns as the others:

> How to deploy the AWS Lambda Power Tuning using the CDK for Java

File: cdk/java/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+This is a blank project for CDK development with Java.
```
Comment (alexcasalboni) [inline cdk/java/README.md:3]: Is it? 😆

File: cdk/java/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+## Useful commands
+
+ * `mvn package`     compile and run tests
+ * `cdk ls`          list all stacks in the app
+ * `cdk synth`       emits the synthesized CloudFormation template
+ * `cdk deploy`      deploy this stack to your default AWS account/region
+ * `cdk diff`        compare deployed stack with current state
+ * `cdk docs`        open CDK documentation
+
+Enjoy!
```
Comment (alexcasalboni) [inline cdk/java/README.md:18]: I'd remove

File: cdk/java/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+This is a blank project for CDK development with Java.
+
+The `cdk.json` file tells the CDK Toolkit how to execute your app.
+
+It is a [Maven](https://maven.apache.org/) based project, so you can open this project with any Maven compatible Java IDE to build and run tests.
+
+## Useful commands
+
+ * `mvn package`     compile and run tests
```
Comment (alexcasalboni) [inline cdk/java/README.md:11]: Here I would only include the commands for building, testing, and deploying the stack (while the generic CDK commands are already documented [here](https://github.com/alexcasalboni/aws-lambda-power-tuning/tree/master/cdk))

File: cdk/java/README.md (hunk 2 above)
Comment (alexcasalboni) [reply cdk/java/README.md:3]: Please check the `Language specific prerequisites` section we have in the other patterns (Go, Python, TypeScript, etc.)

File: cdk/java/src/main/java/com/myorg/JavaStack.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.sam.CfnApplication;
+import software.amazon.awscdk.services.sam.CfnApplication.ApplicationLocationProperty;
+import java.util.HashMap;
+import java.util.Map;
+
+public class JavaStack extends Stack {
```
Comment (alexcasalboni) [inline cdk/java/src/main/java/com/myorg/JavaStack.java:12]: Should this class have a more specific name? Or is this a default we usually recommend for Java-based CDK projects?

See the C# pattern for example: https://github.com/alexcasalboni/aws-lambda-power-tuning/blob/master/cdk/csharp/src/Cdk/TheLambdaPowerTunerStack.cs

**Review by alexcasalboni — CHANGES_REQUESTED** [review]
Awesome 🚀  thanks for contributing 🙏  just a few requested changes

File: cdk/java/src/main/java/com/myorg/JavaStack.java (hunk 5 above)
Comment (patmeiler) [reply cdk/java/src/main/java/com/myorg/JavaStack.java:12]: It's the standard name, but I can rename so that it in line with the other CDK stacks.

Comment (alexcasalboni) [reply cdk/java/src/main/java/com/myorg/JavaStack.java:12]: ok great 🚀

**Pull request merged**